src/components/Recipes/RecipeItem.tsx

```diff
@@ -26,29 +26,42 @@ const RecipeItem: React.FC<{recipeID: number, text: string, image: string, }> =
 
   const favoriteClickHandler = async (event: React.FormEvent<HTMLFormElement>) => {
     event.preventDefault();
-    addFavorite(props.text, props.recipeID, props.image);
-
-    fetchFormat('http://localhost:4000/favoriteAdd', 'POST', {
-      id: await fetchID(),
-      recipe: {
-        title: props.text,
-        id: props.recipeID,
-        image: props.image
-      }
-    });
+    if (!props.isFavorite) {
+      addFavorite(props.text, props.recipeID, props.image);
+      fetchFormat('http://localhost:4000/favoriteAdd', 'POST', {
```

Comment:
you can separate out this body in a const, and reuse it in both conditions instead of repeating yourself